### PR TITLE
`Development`: Fix IrisMessageIntegrationTest for MySQL

### PR DESCRIPTION
**Pull request by maximiliansoelch**
> [!WARNING]
> **This PR includes a migration! Do not deploy to a normal test server.**

#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).

### Motivation and Context

Currently, `sendMessageRateLimitReached()` is flaky and fails when run against MySQL.
This can also cause the succeeding tests to fail because the rate limits are not reset properly.

### Description

This PR adds a migration to change `datetime` to `datetime(3)` in three tables to avoid issues with cut-off dates on MySQL (similar to #6046).

In addition, it adds an entry for the recommended type in the documentation: https://artemis-platform--7645.org.readthedocs.build/en/7645/dev/migration/structure.html#datetime

### Steps for Testing

TBD

#### Code Review
- [x] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### CODE REVIEW

File: src/main/resources/config/liquibase/changelog/20231121165300_changelog.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        <!-- Introduced in 20230705141500_changelog.xml -->
+        <modifyDataType tableName="iris_session" columnName="creation_date" newDataType="datetime(3)"/>
+        <!-- Introduced in 20230714012100_changelog.xml -->
+        <modifyDataType tableName="jhi_user" columnName="iris_accepted" newDataType="datetime(3)"/>
+
+        <!--
+        Due to mysql SQL limitations, modifyDataType will lose primary key/autoincrement/not null/comment settings explicitly redefined in the change.
+        Therefore, we have to explicitly set not null again for the affected columns!
+        -->
+        <addNotNullConstraint tableName="iris_session" columnName="creation_date" columnDataType="datetime(3)"/>
```
Comment (Hialus) [inline src/main/resources/config/liquibase/changelog/20231121165300_changelog.xml:15]: Have you also tested this with Postgres?

**Review by Hialus — CHANGES_REQUESTED** [review]
Changes look good to me.
But I'd suggest that you add this to the documentation, so that the right type is always used in the future.

File: src/main/resources/config/liquibase/changelog/20231121165300_changelog.xml (hunk 1 above)
Comment (maximiliansoelch) [reply src/main/resources/config/liquibase/changelog/20231121165300_changelog.xml:15]: Relevant SO post: https://stackoverflow.com/a/75735575

**Review by Hialus — APPROVED** [review]
Changes look good to me. 
I sadly can't test it locally right now, as my DB contains the Iris settings migration, which requires the accompanying PR for Iris to work. Same goes for TS9.

**Review by krusche — APPROVED** [review]

**Pull request merged**